homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -201,6 +233,21 @@ class XiaomiMiioSwitchDescription(SwitchEntityDescription):
         method_on="async_set_led_on",
         method_off="async_set_led_off",
     ),
+    XiaomiMiioSwitchDescription(
+        key=ATTR_LEARN_MODE,
+        feature=FEATURE_SET_LEARN_MODE,
+        name="Learn Mode",
+        icon="mdi:school-outline",
+        method_on="async_set_learn_mode_on",
+        method_off="async_set_learn_mode_off",
+    ),
+    XiaomiMiioSwitchDescription(
+        key=ATTR_AUTO_DETECT,
+        feature=FEATURE_SET_AUTO_DETECT,
+        name="Auto Detect",
```

Comment:
I have no idea which icon to use for `auto detect`. Honestly I don't know what the `auto detect` does :)

Reply:
@rytilahti I'm not sure if `zhimi.airpurifier.v6` supports `auto_detect` - according to [this](https://github.com/rytilahti/python-miio/blob/master/miio/airpurifier.py#L53-L60). [Here](https://github.com/rytilahti/python-miio/blob/master/miio/airpurifier.py#L492) we have an information that only V1 and V2 support it. Could you help?

Reply:
That might be just a stale comment and other models may also support it. Do you have insight on this, @syssi?

Reply:
I don't know any details about this feature.

Reply:
I've found some insights at the android app of the `zhimi.airpurifier.v1` and `zhimi.airpurifier.v3`: 

```
    <string name="air_purifier_act_det_desc">Mi Air Purifier can monitor air quality when it\'s off. The process will consume a tiny amount of power, and you may hear the motor working. Turn on?</string>
    <string name="air_purifier_act_det_fail">Couldn\'t turn on</string>
    <string name="air_purifier_act_det_no">Not now</string>
    <string name="air_purifier_act_det_processing">Working…</string>
    <string name="air_purifier_act_det_success">Turned on. You can turned this feature off in Settings.</string>
    <string name="air_purifier_act_det_yes">Turn on</string>
    <string name="air_purifier_indoor_act_det">Real time AQI monitor &gt;</string>
    <string name="air_purifier_setting_act_det">Real-time AQI monitor</string>
    <string name="air_purifier_setting_act_det_desc">Requires a tiny amount of power daily</string>
```

Reply:
It looks like the feature is supported of the `zhimi.airpurifier.v1` only (if we ignore the MCU version check): 

```
    public boolean isAutoDetectSupported() {
        if (isV2Model() || isV6Model() || isM2SModel() || isSuperModel()) {
            return false;
        }
        if (!AirPurifierUtil.V1MODEL.equalsIgnoreCase(getModel())) {
            return true;
        }
        return this.mcu_version > 0 && this.mcu_version >= 1126;
    }
```